youtube_dl/extractor/tistory.py

```diff
@@ -0,0 +1,228 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import (
+    url_basename,
+    mimetype2ext,
+    HEADRequest,
+    ExtractorError
+)
+from ..compat import (
+    compat_urllib_request,
+    compat_urllib_error,
+    compat_urlparse,
+    compat_str
+)
+
+import os.path
+import cgi
+import re
+
+
+class TistoryBaseIE(InfoExtractor):
+    _TI_MEDIA_URL = r'https?://cfile[0-9]*.uf.(tistory.com|daum.net)/(?:media|attach|attachment|original)/(?P<id>[A-Za-z0-9]*)'
+
+    def _ti_unquote(self, url):
+        return compat_urlparse.unquote(url)
+
+    def _ti_get_title(self, url, response):
+        _, params = cgi.parse_header(response.info().get('Content-Disposition', ''))
+        if "filename" in params:
+            filename = params["filename"]
+        else:
+            filename = url_basename(url)
+
+        retval = os.path.splitext(self._ti_unquote(filename))[0]
+
+        if type(retval) != compat_str:
```

Comment:
filename should already be a unicode object. Does cgi.parse_header return bytes-like objects? If so a wrapper is necessary.